platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1507,7 +1507,25 @@ - (NSExpression *)mgl_expressionLocalizedIntoLocale:(nullable NSLocale *)locale
                         localizedKeyPath = [NSString stringWithFormat:@"name_%@", preferredLanguage];
                     }
                 }
-                return [NSExpression expressionForKeyPath:localizedKeyPath];
+                // If the keypath is `name`, no need to fallback
+                if ([localizedKeyPath isEqualToString:@"name"]) {
+                    return [NSExpression expressionForKeyPath:localizedKeyPath];
+                }
+                // If the keypath is `name_zh-Hans`, fallback to `name_zh` to `name`
+                // The `name_zh-Hans` field was added since Mapbox Streets v7
+                // See the documentation of name fields for detail https://www.mapbox.com/vector-tiles/mapbox-streets-v7/#overview
+                // CN tiles might using `name_zh-CN` for Simplified Chinese
+                if ([localizedKeyPath isEqualToString:@"name_zh-Hans"]) {
+                    return [NSExpression expressionWithFormat:@"mgl_coalesce({%K, %K, %K, %K})",
+                            localizedKeyPath, @"name_zh-CN", @"name_zh", @"name"];
+                }
+                // Mapbox Streets v8 has `name_zh-Hant`, we should fallback to Simplified Chinese if the filed has no value
```

Comment:
Note that label localization is currently disabled for Streets source v8 because this code only recognizes v6 and v7:

https://github.com/mapbox/mapbox-gl-native/blob/cb714d57c5c5ad181aaf5e1690221da8d965682b/platform/darwin/src/MGLVectorTileSource.mm#L144

But this fallback is good to have in preparation for #11867 anyhow.